### PR TITLE
Protobuf docservice handles structs with empty children correctly

### PR DESCRIPTION
**Pull request by jrhee17**
Motivation:

`DocService` can throw an exception if a `protobuf` `Struct` doesn't have any fields. (which is valid)
See https://line-armeria.slack.com/archives/C1NGPBUH2/p1663556941998359 for more details

Modifications:

- Only iterate recursively through structs if children is non-empty

Result:

- `DocService` is displayed correctly.

### CODE REVIEW

**Review by minwoox — APPROVED** [review]
Thanks! 👍

File: protobuf/src/test/proto/messages.proto [hunk 1]
```diff
@@ -150,6 +150,8 @@ message TestMessage {
   message Nested {
     string string = 1;
   }
+  message EmptyNested {
+  }
```
Comment (ikhoon) [inline protobuf/src/test/proto/messages.proto:154]: Oops... I didn't think of this corner case. 😅

File: protobuf/src/main/java/com/linecorp/armeria/server/protobuf/ProtobufNamedTypeInfoProvider.java [hunk 2]
```diff
@@ -129,7 +129,8 @@ private static FieldInfo newFieldInfo(FieldDescriptor fieldDescriptor, Set<Descr
         final TypeSignature typeSignature = newFieldTypeInfo(fieldDescriptor);
         final Object typeDescriptor = typeSignature.namedTypeDescriptor();
         final FieldInfoBuilder builder;
-        if (typeDescriptor instanceof Descriptor && visiting.add((Descriptor) typeDescriptor)) {
+        if (typeDescriptor instanceof Descriptor && visiting.add((Descriptor) typeDescriptor) &&
+            !((Descriptor) typeDescriptor).getFields().isEmpty()) {
```
Comment (ikhoon) [inline protobuf/src/main/java/com/linecorp/armeria/server/protobuf/ProtobufNamedTypeInfoProvider.java:133]: The following validation is pretty annoying and I'm not sure it is genuinely useful validation.
https://github.com/line/armeria/blob/755c28292c734e954bd69aebcfaeabcf7b024c16/core/src/main/java/com/linecorp/armeria/server/docs/FieldInfoBuilder.java#L56-L57

I was wondering if we could solve the problem by just removing the validation.

Comment (jrhee17) [reply protobuf/src/main/java/com/linecorp/armeria/server/protobuf/ProtobufNamedTypeInfoProvider.java]: I didn't really take a close look but I guess so 😅
